File: SAE6/jardin_de_cocagne/lib/models/basket.dart
// lib/models/basket.dart
import 'package:jardin_de_cocagne/models/basket_product.dart';

class Basket {
  final String id; // ID sous forme de chaîne de caractères
  final String name;
  final double price;
  final String description;
  final String imageUrl;
  final String weight;
  final List<BasketProduct> products;

  Basket({
    required this.id,
    required this.name,
    required this.price,
    required this.description,
    required this.imageUrl,
    required this.weight,
    required this.products,
  });

  // Méthode pour obtenir l'ID sous forme d'entier
  int get idAsInt {
    return int.tryParse(id) ?? 0;
  }

  factory Basket.fromJson(Map<String, dynamic> json) {
    return Basket(
      id: json['id'].toString(), // Convertir en chaîne de caractères
      name: json['name'],
      price: double.parse(json['price'].toString()),
      description: json['description'] ?? '',
      imageUrl: json['image_url'] ?? '',
      weight: json['weight'] ?? '',
      products: json['products'] != null
          ? List<BasketProduct>.from(
              json['products'].map((product) => BasketProduct.fromJson(product)))
          : [],
    );
  }
}

File: SAE6/jardin_de_cocagne_docker/api/src/models/Basket.ts
import { BasketProduct } from './BasketProduct';

export interface Basket {
  id: number;
  name: string;
  price: number;
  description?: string;
  image_url?: string;
  weight?: string;
  is_active: boolean;
  created_at: Date;
  updated_at: Date;
  products?: BasketProduct[];
}

export interface CreateBasketDTO {
  name: string;
  price: number;
  description?: string;
  image_url?: string;
  weight?: string;
  is_active?: boolean;
}

export interface UpdateBasketDTO {
  name?: string;
  price?: number;
  description?: string;
  image_url?: string;
  weight?: string;
  is_active?: boolean;
}